src/Functions/JSONPath/Generators/GeneratorJSONPath.h

```diff
@@ -0,0 +1,127 @@
+#pragma once
+
+#include <Functions/JSONPath/Generators/IGenerator.h>
+#include <Functions/JSONPath/Generators/VisitorJSONPathMemberAccess.h>
+#include <Functions/JSONPath/Generators/VisitorJSONPathRange.h>
+#include <Functions/JSONPath/Generators/VisitorJSONPathRoot.h>
+#include <Functions/JSONPath/Generators/VisitorJSONPathStar.h>
+#include <Functions/JSONPath/Generators/VisitorStatus.h>
+
+#include <Functions/JSONPath/ASTs/ASTJSONPath.h>
+
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int LOGICAL_ERROR;
+}
+
+template <typename JSONParser>
+class GeneratorJSONPath : public IGenerator<JSONParser>
+{
+public:
+    /**
+     * Traverses children ASTs of ASTJSONPathQuery and creates a vector of corresponding visitors
+     * @param query_ptr_ pointer to ASTJSONPathQuery
+     */
+    GeneratorJSONPath(ASTPtr query_ptr_)
+    {
+        query_ptr = query_ptr_;
+        const auto * path = query_ptr->as<ASTJSONPath>();
+        if (!path)
+        {
+            throw Exception("Invalid path", ErrorCodes::LOGICAL_ERROR);
+        }
```

Comment:
Looks like not a good exception message.

Reply:
decided to leave as is (in DMs)